File: backend/src/domain/entities/masters/test_master.py
"""TestMaster — domain entity."""

from dataclasses import dataclass, field

from src.domain.entities.base_entity import BaseEntity


@dataclass
class TestMaster(BaseEntity):
    """Test master linked to a Product."""

    test_name: str = field(default="")
    sample_description: str = field(default="")
    sample_qty: int = field(default=0)
    product_id: int = field(default=0)
    is_active: bool = field(default=True)

File: frontend/src/features/masters/models/TestMaster.ts
import type { MasterListParams, MasterRecord } from './common';

export interface TestMaster extends MasterRecord {
  test_name: string;
  sample_description: string;
  sample_qty: number;
  product_id: number;
}

export interface CreateTestMasterRequest {
  test_name: string;
  sample_description: string;
  sample_qty: number;
  product_id: number;
  is_active: boolean;
}

export type UpdateTestMasterRequest = Partial<CreateTestMasterRequest>;

export interface TestMasterListParams extends MasterListParams {
  product_id?: number;
}
